packages/google-compute-engine-oslogin/utils/oslogin_utils.cc

```diff
@@ -470,6 +469,105 @@ bool ParseJsonToPasswd(const string& json, struct passwd* result,
   return ValidatePasswd(result, buf, errnop);
 }
 
+bool ParseJsonToGroup(const string& json, struct group* result,
```

Comment:
the response can contain key = "next_page_token" which is the pagination token to retrieve next set of results. Same for GroupUsers method

Reply:
updated. this resulted in a significant rewrite.